src/mam4xx/hetfrz.hpp

```diff
@@ -0,0 +1,1490 @@
+  fn_bc_c1_num = 0;
+  auto &fn_dst_c1_num = diags.fn_dst_c1_num(k);
+  fn_dst_c1_num = 0;
+  auto &fn_dst_c3_num = diags.fn_dst_c3_num(k);
+  fn_dst_c3_num = 0;
+  auto &na500 = diags.na500(k);
+  na500 = 0;
+  auto &totna500 = diags.totna500(k);
+  totna500 = 0;
+  auto &freqimm = diags.freqimm(k);
+  freqimm = 0;
+  auto &freqcnt = diags.freqcnt(k);
+  freqcnt = 0;
+  auto &freqdep = diags.freqdep(k);
+  freqdep = 0;
+  auto &freqmix = diags.freqmix(k);
+  freqmix = 0;
+  auto &dstfrezimm = diags.dstfrezimm(k);
+  dstfrezimm = 0;
+  auto &dstfrezcnt = diags.dstfrezcnt(k);
+  dstfrezcnt = 0;
+  auto &dstfrezdep = diags.dstfrezdep(k);
+  dstfrezdep = 0;
+  auto &bcfrezimm = diags.bcfrezimm(k);
+  bcfrezimm = 0;
+  auto &bcfrezcnt = diags.bcfrezcnt(k);
+  bcfrezcnt = 0;
+  auto &bcfrezdep = diags.bcfrezdep(k);
+  bcfrezdep = 0;
+  auto &nimix_imm = diags.nimix_imm(k);
+  nimix_imm = 0;
+  auto &nimix_cnt = diags.nimix_cnt(k);
+  nimix_cnt = 0;
+  auto &nimix_dep = diags.nimix_dep(k);
+  nimix_dep = 0;
+  auto &dstnidep = diags.dstnidep(k);
+  dstnidep = 0;
+  auto &dstnicnt = diags.dstnicnt(k);
+  dstnicnt = 0;
+  auto &dstniimm = diags.dstniimm(k);
+  dstniimm = 0;
+  auto &bcnidep = diags.bcnidep(k);
+  bcnidep = 0;
+  auto &bcnicnt = diags.bcnicnt(k);
+  bcnicnt = 0;
+  auto &bcniimm = diags.bcniimm(k);
+  bcniimm = 0;
+  auto &numice10s = diags.numice10s(k);
+  numice10s = 0;
+  auto &numimm10sdst = diags.numimm10sdst(k);
+  numimm10sdst = 0;
+  auto &numimm10sbc = diags.numimm10sbc(k);
+  numimm10sbc = 0;
+  // End fields used for diagnostics.
+
+  const int coarse_idx = int(ModeIndex::Coarse);
+  const int accum_idx = int(ModeIndex::Accumulation);
+  const int pcarbon_idx = int(ModeIndex::PrimaryCarbon);
+
+  auto &accum_bc = progs.q_aero_i[accum_idx][int(AeroId::BC)];
+  auto &accum_dst = progs.q_aero_i[accum_idx][int(AeroId::DST)];
+  auto &accum_soa = progs.q_aero_i[accum_idx][int(AeroId::SOA)];
+  auto &accum_so4 = progs.q_aero_i[accum_idx][int(AeroId::SO4)];
+  auto &accum_pom = progs.q_aero_i[accum_idx][int(AeroId::POM)];
+  auto &accum_mom = progs.q_aero_i[accum_idx][int(AeroId::MOM)];
+  auto &num_accum = progs.n_mode_i[accum_idx];
+
+  auto &pcarbon_bc = progs.q_aero_i[pcarbon_idx][int(AeroId::BC)];
+  auto &pcarbon_pom = progs.q_aero_i[pcarbon_idx][int(AeroId::POM)];
+  auto &pcarbon_mom = progs.q_aero_i[pcarbon_idx][int(AeroId::MOM)];
+
+  auto &coarse_dust = progs.q_aero_i[coarse_idx][int(AeroId::DST)];
+  auto &coarse_ncl = progs.q_aero_i[coarse_idx][int(AeroId::NaCl)];
+  auto &coarse_mom = progs.q_aero_i[coarse_idx][int(AeroId::MOM)];
+  auto &coarse_bc = progs.q_aero_i[coarse_idx][int(AeroId::BC)];
+  auto &coarse_pom = progs.q_aero_i[coarse_idx][int(AeroId::POM)];
+  auto &coarse_soa = progs.q_aero_i[coarse_idx][int(AeroId::SOA)];
+  auto &coarse_so4 = progs.q_aero_i[coarse_idx][int(AeroId::SO4)];
+
+  auto &num_coarse = progs.n_mode_i[coarse_idx];
+
+  // initialize rho
+  std::cout.precision(17);
```

Comment:
@pressel here is your CI bug. Only happens on double precision for some reason, but a valid failure

Reply:
Thanks for tracking this down @CameronRutherford, somehow I missed it when looking through the CI Logs.